docs/docs/data-fetching.md

```diff
@@ -0,0 +1,221 @@
+---
+title: Build Time and Client Runtime Data Fetching
+---
+
+import BuildDataExample from "../../www/src/components/build-data-example.js"
+import ClientDataExample from "../../www/src/components/client-data-example.js"
+
+This guide demonstrates how to fetch data at both [_build time_](/docs/glossary#build) and [_runtime_](/docs/glossary#runtime) in Gatsby.
```

Comment:
What do you think about adding this kind of intro statement to set expectations, so that people don't go down a rabbit-hole before realizing there's already a plugin for their use case?
```suggestion
This guide demonstrates how to fetch data at both [_build time_](/docs/glossary#build) and [_runtime_](/docs/glossary#runtime) in Gatsby. Most of the techniques outlined are for custom data handling. Be sure to check out Gatsby's [plugin library](/plugins/) to see if there's an off-the-shelf solution for your data requirements, such as [sourcing from a CMS](/docs/headless-cms/) or other third-party integration.
```